_content/v2-preview/java/_quick-start.md

```diff
@@ -0,0 +1,281 @@
+# Quick Start
+This guide will walk you through setting up a Java project that listens to device activations and messages and responds to every 3rd message.
+
+> This guide assumes the sketch and payload functions of [The Things Uno / Quick Start](/uno/#quick-start), but can be easily applied to any other.
+
+The full script that we will build is also included as [example](https://github.com/TheThingsNetwork/java-app-lib/blob/master/sample/src/main/java/org/thethingsnetwork/java/app/sample/Test.java) in the Java library.
+
+## Setup
+Let's install Java, create a Java project and require the TTN Client.
+
+1.  [Download](https://www.java.com/fr/download/) and install Java. Under debian-based systems, use :
```

Comment:
Mac comes with Oracle Java 1.6 minimum, so if you think it works with that old version you can mention Mac has it pre-installed. Otherwise for Mac and Windows just point to https://www.java.com/download/ (remove the `/fr`)